src/Optimizer/ImageDimensions.php

```diff
@@ -119,23 +129,46 @@ public function isTiny($threshold = self::TINY_THRESHOLD)
             }
         }
 
+        // If one or both of the dimensions are missing, we cannot deduce an aspect ratio.
         if ($this->getWidth() === null || $this->getHeight() === null) {
             return true;
         }
 
+        // If one or both of the dimensions are zero, the entire image will be invisible.
         if (
             (is_numeric($this->getWidth()) && $this->getWidth() <= 0)
             || (is_numeric($this->getHeight()) && $this->getHeight() <= 0)
         ) {
             return true;
         }
 
+        // If relative units are in use, we cannot assume much about the final dimensions.
+        if (
+            (
+                $this->hasWidthUnit()
+                && in_array($this->getWidthUnit(), LengthUnit::RELATIVE_UNITS, true)
+                && (is_numeric($this->getHeight()) && $this->getHeight() < $threshold)
+            ) || (
+                $this->hasHeightUnit()
+                && in_array($this->getHeightUnit(), LengthUnit::RELATIVE_UNITS, true)
+                && (is_numeric($this->getWidth()) && $this->getWidth() < $threshold)
+            )
+        ) {
+            return false;
+        }
```

Comment:
It this missing a check to see if _both_ the `width` and `height` are relative?
```suggestion
        // If relative units are in use, we cannot assume much about the final dimensions.
        $isRelativeWidth = in_array($this->getWidthUnit(), LengthUnit::RELATIVE_UNITS, true);
        $isRelativeHeight = in_array($this->getHeightUnit(), LengthUnit::RELATIVE_UNITS, true);
        if (
            (
                $isRelativeWidth && $isRelativeHeight
            ) || (
                $isRelativeWidth
                && (is_numeric($this->getHeight()) && $this->getHeight() < $threshold)
            ) || (
                $isRelativeHeight
                && (is_numeric($this->getWidth()) && $this->getWidth() < $threshold)
            )
        ) {
            return false;
        }
```

This could be optimized a bit better to

Reply:
Text seems cut off here. I'm not sure what this is supposed to be improve, as the first condition would already be triggered in that case as well (without doing both of the calculations).

Reply:
Ah, I assume you mean that this would allow to fail without doing the comparison against the threshold.

If so, I would assume that the most costly operation here is actually the `in_array()`. If we'd want to further optimize this piece of logic, we should first look into turning this into a hashed lookup via `array_key_exists()` instead.

Reply:
Sorry for the truncation.

If not for the optimization, what seems to me missing is adding a first condition to check if _both_ the width and height are relative. Or is that accounted for?

Reply:
This is what I'm referring to: https://github.com/ampproject/amp-toolbox-php/pull/49/files#r541983412,

Reply:
Ah, yes, I got what you mean now. The two conditions only work exclusively on a single relative unit as the other needs to be numeric. I'll adapt.

Reply:
Resolved via https://github.com/ampproject/amp-toolbox-php/pull/49/commits/b76f3fce804115b085e7b50702bafb901dd776a9